File: skillful_network_server/src/main/java/fr/uca/cdr/skillful_network/entities/application/Training.java
package fr.uca.cdr.skillful_network.entities.application;

import javax.persistence.*;

import fr.uca.cdr.skillful_network.entities.Keyword;

import java.util.Date;
import java.util.Objects;
import java.util.Set;

@Entity
public class Training {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	private String name;

	private String organization;

	private String description;

	private Date dateBeg;

	private Date dateEnd;

	private Date dateUpload;

	private long durationHours;

	@ManyToMany(fetch = FetchType.EAGER, cascade =  CascadeType.PERSIST)
	private Set<Keyword> keywords;

	@OneToMany(mappedBy = "offer", cascade = CascadeType.ALL, fetch = FetchType.EAGER)
	private Set<TrainingApplication> trainingApplicationSet;

	public Training() {

	}

	public Training(String name,
					String organization,
					String description,
					Date dateBeg,
					Date dateEnd,
					Date dateUpload,
					Long durationHours,
					Set<Keyword> keywords,
					Set<TrainingApplication> trainingApplicationSet) {
		this.name = name;
		this.organization = organization;
		this.description = description;
		this.dateBeg = dateBeg;
		this.dateEnd = dateEnd;
		this.dateUpload = dateUpload;
		this.durationHours = durationHours;
		this.keywords = keywords;
		this.trainingApplicationSet = trainingApplicationSet;
	}

	public Long getId() {
		return id;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getOrganization() {
		return organization;
	}

	public void setOrganization(String organization) {
		this.organization = organization;
	}

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public Date getDateBeg() {
		return dateBeg;
	}

	public void setDateBeg(Date dateBeg) {
		this.dateBeg = dateBeg;
	}

	public Date getDateEnd() {
		return dateEnd;
	}

	public void setDateEnd(Date dateEnd) {
		this.dateEnd = dateEnd;
	}

	public Date getDateUpload() {
		return dateUpload;
	}

	public void setDateUpload(Date dateUpload) {
		this.dateUpload = dateUpload;
	}

	public long getDurationHours() {
		return durationHours;
	}

	public void setDurationHours(long durationHours) {
		this.durationHours = durationHours;
	}

	public Set<Keyword> getKeywords() {
		return keywords;
	}

	public void setKeywords(Set<Keyword> keywords) {
		this.keywords = keywords;
	}

	public Set<TrainingApplication> getTrainingApplicationSet() {
		return trainingApplicationSet;
	}

	public void setTrainingApplicationSet(Set<TrainingApplication> trainingApplicationSet) {
		this.trainingApplicationSet = trainingApplicationSet;
	}

	@Override
	public boolean equals(Object o) {
		if (this == o) return true;
		if (o == null || getClass() != o.getClass()) return false;
		Training training = (Training) o;
		return Objects.equals(id, training.id);
	}

	@Override
	public int hashCode() {
		return Objects.hash(id);
	}
}

File: skillful_network_client/src/app/shared/models/application/training.ts
import { TrainingApplication } from './training-application';

export class Training {

    private _id: number;
    private _name: string;
    private _organization: string;
    private _description: string;
    private _dateBeg: Date;
    private _dateEnd: Date;
    private _durationHours: number;
    private _dateUpload: Date;
    private _keywords: string[];
    private _trainingApplications: TrainingApplication[];

    constructor(data: any) {
        this.id = data.id;
        this.name = data.name;
        this.organization = data.organization;
        this.description = data.description;
        this.dateBeg = data.dateBeg;
        this.dateEnd = data.dateEnd;
        this.durationHours = data.durationHours;
        this.dateUpload = data.dateUpload;
        this.keywords = data.keywords;
        this.trainingApplications = data.trainingApplicationSet;
    }

    public set trainingApplications(trainingApplications : TrainingApplication[]) {
        this._trainingApplications = trainingApplications;
    }

    public get trainingApplications() : TrainingApplication[] {
        return this._trainingApplications;
    }

    public get organization() : string {
        return this._organization;
    }
    
    public set organization(organization: string) {
        this._organization = organization;
    }

    get id(): number {
        return this._id;
    }

    set id(value: number) {
        this._id = value;
    }

    get description(): string {
        return this._description;
    }

    set description(value: string) {
        this._description = value;
    }

    get keywords(): string[] {
        return this._keywords;
    }

    set keywords(value: string[]) {
        this._keywords = value;
    }

    get dateUpload(): Date {
        return this._dateUpload;
    }

    set dateUpload(value: Date) {
        this._dateUpload = value;
    }

    get durationHours(): number {
        return this._durationHours;
    }

    set durationHours(value: number) {
        this._durationHours = value;
    }

    get dateEnd(): Date {
        return this._dateEnd;
    }

    set dateEnd(value: Date) {
        this._dateEnd = value;
    }

    get name(): string {
        return this._name;
    }

    set name(value: string) {
        this._name = value;
    }

    get dateBeg(): Date {
        return this._dateBeg;
    }

    set dateBeg(value: Date) {
        this._dateBeg = value;
    }

}